docs/certificates-for-localhost.md

```diff
@@ -0,0 +1,125 @@
+top_graphic: 1
+date: 2017-12-21T00:00
+---
+
+Last updated: {{ page.date | date: "%B %d, %Y" }} \| [See all Documentation](/docs/)
+
+Sometimes people want to get a certificate for the hostname "localhost", either
+for use in local development, or for distribution with a native application that
+needs to communicate with a web application. Let's Encrypt can't provide
+certificates for "localhost" because nobody uniquely owns it, and it's not
+rooted in a top level domain like ".com" or ".net". It's possible to
+set up your own domain name that happens to resolve to 127.0.0.1, and get a
+certificate for it using the DNS challenge. However, this is generally a bad
+idea and there are better options.
+
+# For local development
+
+If you're developing a web app, it's useful to run a local web server like
+Apache or Nginx, and access it via http://localhost:8000/ in your web browser.
+However, web browsers behave in subtly different ways on HTTP vs HTTPS pages.
+The main difference: On an HTTPS page, any requests to load JavaScript from an
+HTTP URL will be blocked. So if you're developing locally using HTTP, you might
+add a script tag that works fine on your development machine, but breaks when
+you deploy to your HTTPS production site. To catch this kind of problem, it's
+useful to set up HTTPS on your local web server. However, you don't want to see
+certificate warnings all the time. How do you get the green lock locally?
+
+The best option: Generate your own certificate, either self-signed or signed by
+a local root, and trust it in your operating system's trust store. Then use that
+certificate in your local web server. See below for details.
+
+# For native apps talking to web apps
+
+Sometimes developers want to offer a downloadable native app that can be
+used alongside a web site to offer extra features. For instance, the Dropbox
+and Spotify desktop apps scan for files from across your machine, which a
+web app would not be allowed to do. One common approach is for these native
+apps to offer a web service on localhost, and have the web app make requests
+to it via XMLHTTPRequest (XHR) or WebSockets. The web app almost always uses HTTPS, which
+means that browsers will forbid it from making XHR or WebSockets requests
+to non-secure URLs. This is called Mixed Content Blocking. To communcate with
+the web app, the native app needs to provide a secure web service.
+
+Fortunately, modern browsers [consider][mcb-localhost] "http://127.0.0.1:8000/" to be a
+["potentially trustworthy"][secure-contexts]
+URL because it refers to a loopback address. Traffic sent to 127.0.0.1 is guaranteed
+not to leave your machine, and so is considered automatically secure against
+network interception. That means if your web app is HTTPS, and you offer a
+native app web service on 127.0.0.1, the two can happily communicate via XHR.
+Unfortunately, [localhost currently doesn't yet get the same treatment][let-localhost].
```

Comment:
"currently doesn't yet" reads a little awkwardly to me. How about "at present localhost doesn't get the same treatment"?